mentorship/gsod/2022/project-ideas.md

```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
+
+- Installation Guidelines for Keptn and Keptn CLI client on common Kubernetes clusters.
+  This deliverable will update or replace the existing documentation.
+- New example projects that demonstrate key Keptn capabilities to newcomer users and potential adopters.
+  This deliverable will update or replace the existing documentation and tutorials.
+- Restructure existing documentation to user and administrator guidelines
+  and developer guides.
+- Reference pages for key information and concepts that are not specific to Keptn versions.
+
+### Details for each deliverable
+
+#### Installation Guidelines
+
+Create and test installation instructions for creating a k8s cluster on which Keptn can be installed for demo and evaluation purposes.
+Create excellent instructions for installing Keptn CLI and Keptn on this new cluster.
+Confirm that one set of instructions works for all k8s implementations and all “Keptn curious” projects we provide. 
+If necessary, create multiple instruction sets, clearly defining when each is appropriate.
+
+Also, we propose to create and test instructions for installing a Keptn “playground” in an existing k8s cluster for study and demos. 
+The goal here is to provide an environment where someone can try Keptn out with their existing data without risking damage to an existing production system (such as Prometheus, Dynatrace, DataDog, etc.)
+
+For this deliverable, instructions should be pedagogical rather than pure cookbook:
+provide enough explanation that the user gains some understanding of what they are doing.
+Include information for verifying each step and some troubleshooting information or links.
+
+#### New example projects
+
+Create 5-6 projects that introduce key Keptn capabilities for the Keptn curious to explore. 
+We have a number of existing tutorials to use as starters or inspiration. 
+Note that many of the existing tutorials use the [sockshop project](https://github.com/keptn-demo/sockshop) as a base;
+and some updates will be needed there.
+
+Each project should include:
+
+- Files that can be downloaded from github to provide some basic configuration with an explanation of what is provided in these files (that they would have to do themselves if they wanted to implement this functionality.
+- Specification in GitHub for what needs to be installed to provide a viable environment for the project, including integrations that must be included, configurations that must be set, and so forth.
+- Excellent information about how to run the sample project and do the exercises, including context for each step, what to look for to determine success and understand the output.  The goal is to begin to educate the user about how to get the most out of Keptn
+
+#### Separate guides for roles
+
+We want to introduce separate documentation for different Keptn user roles: 
+administrators, end users and developers.
+We suggest to work together to regroup the content so that the new structure is introduced.
```

Comment:
I like this idea but some people on the mentors mailing list have talked about how important it is that each participant's work is completely independent of the work of all other participants.  Will the GSoC organizers see this as running afoul of that guideline?  What if this said, "Each participant is expected to contribute to the design of the new structure and then implement an assigned portion of content for the new structure" or something?

Reply:
>Each participant is expected to contribute to the design of the new structure and then implement an assigned portion of content for the new structure

Yes, I echo that. As per GSoC guidelines, it has to be contributed individually to the assigned project, but not to work together on the same project.

Reply:
It's a delicate subject because, of course, collaboration is so essential to all successful open source projects.  I'm thinking that, by stating that everyone contributes to the plan, it's the plan that may suffer if a participant fails to deliver their chunk of the materail but this would not leave another participant unable to proceed with their work.

Reply:
I suggest rewording in a follow-up pull request. I need to focus on parenting, and I may not be available until the deadline